### ISSUE TITLE
FN S1449: DateTimeOffset, DateOnly and TimeOnly are not supported 

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#6439]
### Description

The rule does not raise an issue for DateTimeOffset, DateOnly, and TimeOnly objects call `ToString()`.

### Repro steps

```
  var s = string.Empty;
  s = new DateTime().ToString(); // FN
  s = new DateTimeOffset().ToString(); // FN
  s = new TimeOnly().ToString(); // FN
  s = new DateOnly().ToString(); // FN
```

### Expected behavior

An issue for S1449 should be raised.

### Related information

* C#/VB.NET Plugins version: 8.49

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1800](https://sonarsource.atlassian.net/browse/NET-1800)

[NET-1800]: https://sonarsource.atlassian.net/browse/NET-1800?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
